drp.py

```diff
@@ -256,6 +265,67 @@ def handle_error(exc, retry=True):
         sublime.set_timeout_async(connect_background, 0)
 
 
+def git_config_parser(path):
+    obj = dict()
+    with open(path) as cfg:
+        lines = cfg.read().split("\n")
+        current_section = None
+
+        for line in lines:
+            # remove comments and spaces
+            line = re.sub(" |;(.*)|#(.*)", "", line)
+            if not line:
+                continue
+
+            if line.startswith("["):
+                res = re.search('"(.*)"', line)
+                if res is not None:
+                    sec_name = re.sub('\[|"(.*)"|\]', "", line)
+                    subsec_name = res.group(1)
+                    if sec_name not in obj:
+                        obj[sec_name] = {}
+
+                    obj[sec_name][subsec_name] = {}
+                    current_section = [sec_name, subsec_name]
+                else:
+                    sec_name = re.sub("\[|\]", "", line)
+                    obj[sec_name] = {}
+                    current_section = [sec_name]
+
+            else:
+                parts = re.sub("\t|\0", "",line).split("=")
+                if len(current_section) < 2:
+                    obj[current_section[0]][parts[0]] = parts[1]
+                else:
+                    obj[current_section[0]][current_section[1]][parts[0]] = parts[1]
+
+    return obj
+
+
+def get_git_url_from_config(folder):
+    gitcfg_path = folder+"/.git/config"
+    if os.path.exists(gitcfg_path):
+        cfg = git_config_parser(gitcfg_path)
+        return cfg["remote"]["origin"]["url"]
+
+    return None
+
+
+def get_git_url(window):
+    for folder in window.folders():
+        url = None
+        try:
+            url = subprocess.check_output(["git", "-C", folder, "remote", "get-url", "origin"], universal_newlines=True)
+        except:
+            url = get_git_url_from_config(folder)
+
+        if url is not None:
+            url = re.sub("\.git\n?$", "", url)
```

Comment:
You'll also need to handle "URL"s like `git@github.com:Snazzah/SublimeDiscordRP.git` (for github; may be different for other services like gitlab and bitbucket).

Reply:
Yes, i was going to change the PR status to draft but i didn't found the option.

Im going to parse the ssh later, from what i saw in gh and gitlab the standard is "git@<domain>:OWNER/REPOSITORY.git"

About the config parser can you point the issues related to that? Or rewriting is better?

Reply:
It *could* be improved but it doesn't need to be. As I mentioned, it should get the job done and I can't think of any edge cases where it would fail, so it's fine to leave it as-is. It does rely a bit on the format of specific config options (e.g. it doesn't work on quoted values) but the url is generally not quoted.